docs/src/content/docs/book/contracts.mdx

```diff
@@ -149,11 +149,42 @@ contract Example {
 }
 ```
 
-State variables must have a default value or initialized in [`init(){:tact}`](#init-function) function, that runs on deployment of the contract. The only exception is persistent state variables of type [`map<K, V>{:tact}`](/book/maps) since they are initialized empty by default.
+State variables must have a default value or be initialized in the [`init(){:tact}`](#init-function) function that runs once on deployment of the contract. The only exception are persistent state variables of type [`map<K, V>{:tact}`](/book/maps), since they are initialized empty by default.
+
+The default value of state variables is assigned before any values could be assigned in the [`init(){:tact}`](#init-function) function.
+
+```tact
+contract Example {
+    // persistent state variables
+    var1: Int = 0; // initialized with default value 0
+
+    // constructor function
+    init() {
+        self.var1 = 42; // overrides the default to 42
+    }
+}
+```
+
+As the contract state is updated at the very end of the [compute phase][compute] of the transaction, intermediate assignments of [`Int{:tact}`][int] values that exceed the limits specified by [serialization formats](/book/integers#serialization) won't fail immediately. Instead, such assignments would cause an [exit code 5](/book/exit-codes#5) only after all statements have been executed.
```

Comment:
Let's add this is especially relevant if the user relies on these out of range errors to cause early termination or uses `dump`. For instance, `self.var = -1; self.var = 0` won't fail at all.